cradle-core/src/main/java/com/exactpro/cradle/CradleStorage.java

```diff
@@ -222,7 +222,10 @@ public void init(boolean prepareStorage) throws CradleStorageException, IOExcept
 		logger.info("Initializing storage");
 		
 		doInit(prepareStorage);
-		bpc = new BookAndPageChecker(getBookCache());
+		// In case it did not get initialized in doInit
+		if (bpc == null) {
```

Comment:
Why this might happen?

Reply:
As bpc is CradleStorage's member that class should try to check whether or not it was initialized. Currently there's no way that can happen, but for sake of consistency and forward-thinking I think its necessary.